docs/resources/Channel.md

```diff
@@ -202,6 +202,7 @@ Represents a message sent in a channel within Discord.
 | application?              | [message application](#DOCS_RESOURCES_CHANNEL/message-object-message-application-structure) object                                              | sent with Rich Presence-related chat embeds                                                                                   |
 | message_reference?        | [message reference](#DOCS_RESOURCES_CHANNEL/message-object-message-reference-structure) object                                                  | reference data sent with crossposted messages                                                                                 |
 | flags?                    | integer                                                                                                                                         | [message flags](#DOCS_RESOURCES_CHANNEL/message-object-message-flags) `OR`d together, describes extra features of the message |
+| referenced_message? | ?[message object](#DOCS_RESOURCES_CHANNEL/message-object) | the message that was replied to. If not present, the backend couldn't fetch the message. If null, the message was deleted. If present and non-null, it is a message object |
```

Comment:
What about when there is not referenced message? Will that fall under `If not present, the backend couldn't fetch the message.`?

Reply:
I assume the intent is to only pay attention to this field if the flag is set, otherwise the flag would be redundant if determinable from the state of this object in the documented behaviour.

It would be nice to get a clear definition of this, though. Even if it duplicates some pattern elsewhere.

Reply:
This field will only appear for replies at the moment.

"Normal" (type 0) messages won't have it, because if they have a `message_reference`, they're crossposts.

Reply:
> What about when its a normal message, not a reply? What will happen with this field? Will that fall under If not present, the backend couldn't fetch the message.?

Fair point. In this case the `nullable` `?` on the field may not be enough, I can clear this up a bit.